rust/agents/relayer/src/msg/gelato_submitter.rs

```diff
@@ -1,98 +1,231 @@
+    pub(crate) sponsor_address: Address,
     /// Interface to agent rocks DB for e.g. writing delivery status upon completion.
-    db: AbacusDB,
+    /// TODO(webbhorn): Promote to non-_-prefixed name once we're checking gas payments.
+    pub(crate) _db: AbacusDB,
+    /// Signer to use for EIP-712 meta-transaction signatures.
+    pub(crate) signer: Signers,
+    /// Shared reqwest HTTP client to use for any ops to Gelato endpoints.
+    /// Intended to be shared by reqwest library.
+    pub(crate) http: reqwest::Client,
+    /// Prometheus metrics.
+    /// TODO(webbhorn): Promote to non-_-prefixed name once we're populating metrics.
+    pub(crate) _metrics: GelatoSubmitterMetrics,
 }
 
 impl GelatoSubmitter {
-    pub fn new(
-        cfg: GelatoConf,
-        rx: mpsc::UnboundedReceiver<SubmitMessageArgs>,
-        inbox_contracts: InboxContracts,
-        db: AbacusDB,
-    ) -> Self {
-        assert!(cfg.enabled_for_message_submission);
-        Self {
-            rx,
-            inbox_contracts,
-            db,
-        }
-    }
-
     pub fn spawn(mut self) -> Instrumented<JoinHandle<Result<()>>> {
         tokio::spawn(async move { self.work_loop().await })
-            .instrument(info_span!("submitter work loop"))
+            .instrument(info_span!("gelato submitter work loop"))
     }
 
-    /// The Gelato relay framework allows us to submit ops in
-    /// parallel, subject to certain retry rules. Therefore all we do
-    /// here is spin forever asking for work from the rx channel, then
-    /// spawn the work to submit to gelato in a root tokio task.
-    ///
-    /// It is possible that there has not been sufficient interchain
-    /// gas deposited in the InterchainGasPaymaster account on the source
-    /// chain, so we also keep a wait queue of ops that we
-    /// periodically scan for any gas updates.
-    ///
-    /// In the future one could maybe imagine also applying a global
-    /// rate-limiter against the relevant Gelato HTTP endpoint or
-    /// something, or a max-inflight-cap on Gelato messages from
-    /// relayers, enforced here. But probably not until that proves to
-    /// be necessary.
     async fn work_loop(&mut self) -> Result<()> {
         loop {
             self.tick().await?;
-            tokio::task::yield_now().await;
+            sleep(Duration::from_millis(1000)).await;
         }
     }
 
-    /// Extracted from main loop to enable testing submitter state
-    /// after each tick, e.g. in response to a change in environment
-    /// conditions like values in InterchainGasPaymaster.
-    async fn tick(&self) -> Result<()> {
-        // TODO(webbhorn): Pull all available messages out of self.rx
-        // and check if enough gas to process them. If not, put on
-        // wait queue. If there is enough, spawn root task and run
-        // the fwd req op.
+    async fn tick(&mut self) -> Result<()> {
+        // Pull any messages sent by processor over channel.
+        loop {
```

Comment:
Confused why the tick function seems to be entirely a loop.

Reply:
Yeah, it's poor naming and you are right to point it out. FWIW this file has totally changed in my work-in-progress branch and doesn't nearly retain this misleading structure. You can check out https://github.com/abacus-network/abacus-monorepo/blob/3cae74f8e65e2d552b79ff9e78df1476a279519e/rust/agents/relayer/src/msg/gelato_submitter.rs if you are curious though I'll be mailing some subset of it for review soon.

I'm inclined to justr leave this here for now since it's dead code and panics in a todo!() immediatley anyway (op.run()), but can delete or replcae it if you prefer. Please let me know, I'll leave this comment unresolved.

Reply:
If it is about to get changed, I have been in the situation of branches off branches before and I just want to make sure it is addressed soon.